File: java/src/model/Mago.java
package model;

// Hereda de Personaje
public class Mago extends Personaje {

    // Constructor
    public Mago(int id, String nombre, int nivel) {

        // llama constructor padre
        super(id, nombre, nivel);
    }

    // Sobrescritura
    @Override
    public void habilidadEspecial() {

        System.out.println("Lanzamiento de hechizo");
    }
}

File: python/model/mago.py
from model.personaje import Personaje

# Herencia
class Mago(Personaje):

    # Constructor
    def __init__(self, id, nombre, nivel):

        # constructor padre
        super().__init__(id, nombre, nivel)

    # sobrescritura
    def habilidad_especial(self):

        print("Lanzamiento de hechizo")